src/stories/Library/disclosure/Disclosure.tsx

```diff
@@ -20,16 +23,26 @@ export const Disclosure: React.FC<DisclosureProps> = ({
           setIsOpen(!isOpen);
         }}
       >
-        <div className="disclosure__icon bg-identity-tint-120 m-24">
-          <img
-            className="disclosure__icon invert"
-            src={`icons/collection/${icon}.svg`}
-            alt="various-icon"
-          />
-        </div>
-        {headline}
+        {!withAvailability && (
+          <div className="disclosure__icon bg-identity-tint-120">
+            <img
+              className="invert"
+              src={`icons/collection/${icon}.svg`}
+              alt="various-icon"
+            />
+          </div>
+        )}
+        {!withAvailability && (
+          <span className="disclosure__headline__text">{headline}</span>
+        )}
+        {withAvailability && (
+          <span className="disclosure__headline__text-shorter">{headline}</span>
```

Comment:
Please rework this BEM structure. You should not have nested elements (headline and text) within your block (disclosure). Modifiers should be separated with `--`.

Reply:
Oh I had no idea that BEM elements don't nest. Great that you point it out :)